modules/pam_limits/pam_limits.c

```diff
@@ -1309,28 +1309,30 @@ pam_sm_open_session (pam_handle_t *pamh, int flags UNUSED,
 
     /* Read subsequent *.conf files, if they exist. */
     char **filename_list = read_limits_dir(pamh);
+    char *last_filename = NULL;
     if (filename_list != NULL) {
         for (i = 0; filename_list[i] != NULL; i++) {
             pl->conf_file = filename_list[i];
             retval = parse_config_file(pamh, pwd->pw_name, pwd->pw_uid, pwd->pw_gid, ctrl, pl, 0);
             if (retval != PAM_SUCCESS)
                 break;
         }
+        strcpy(last_filename, pl->conf_file);
```

Comment:
This one stays in tact though: Since last_filename is NULL, this is a NULL pointer derference.

Reply:
malloc + strcpy can be simplified by strdup. And with this change, last_filename has to be freed later on to avoid memory leakage (error and success paths).